services/dynamic-sidecar/tests/unit/test_core_rabbitmq.py

```diff
@@ -0,0 +1,183 @@
+
+import aio_pika
+import pytest
+from _pytest.fixtures import FixtureRequest
+from fastapi.applications import FastAPI
+from models_library.projects import ProjectID
+from models_library.projects_nodes import NodeID
+from models_library.settings.rabbit import RabbitConfig
+from models_library.users import UserID
+from pytest_mock.plugin import MockerFixture
+from simcore_service_dynamic_sidecar.core.rabbitmq import SLEEP_BETWEEN_SENDS, RabbitMQ
+from simcore_service_dynamic_sidecar.modules import mounted_fs
+
+pytestmark = pytest.mark.asyncio
+
+
+pytest_plugins = [
+    "pytest_simcore.docker_compose",
+    "pytest_simcore.docker_swarm",
+    "pytest_simcore.rabbit_service",
+    "pytest_simcore.repository_paths",
+    "pytest_simcore.tmp_path_extra",
+]
+
+pytest_simcore_core_services_selection = ["rabbit"]
+
+# FIXTURE
+
+
+@pytest.fixture(scope="module")
+def event_loop(request: FixtureRequest) -> Iterator[AbstractEventLoop]:
+    loop = asyncio.get_event_loop_policy().new_event_loop()
+    yield loop
+    loop.close()
+
+
+@pytest.fixture(scope="module")
+def user_id() -> UserID:
+    return 1
+
+
+@pytest.fixture(scope="module")
+def project_id() -> ProjectID:
+    return uuid.uuid4()
+
+
+@pytest.fixture(scope="module")
+def node_id() -> NodeID:
+    return uuid.uuid4()
+
+
+@pytest.fixture(scope="module")
+def mock_environment(
+    event_loop: AbstractEventLoop,
+    mock_dy_volumes: Path,
+    compose_namespace: str,
+    inputs_dir: Path,
+    outputs_dir: Path,
+    state_paths_dirs: List[Path],
+    user_id: UserID,
+    project_id: ProjectID,
+    node_id: NodeID,
+) -> Iterator[None]:
+    with mock.patch.dict(
+        os.environ,
+        {
+            "SC_BOOT_MODE": "production",
+            "DYNAMIC_SIDECAR_COMPOSE_NAMESPACE": compose_namespace,
+            "REGISTRY_AUTH": "false",
+            "REGISTRY_USER": "test",
+            "REGISTRY_PW": "test",
+            "REGISTRY_SSL": "false",
+            "DY_SIDECAR_PATH_INPUTS": str(inputs_dir),
+            "DY_SIDECAR_PATH_OUTPUTS": str(outputs_dir),
+            "DY_SIDECAR_STATE_PATHS": json.dumps([str(x) for x in state_paths_dirs]),
+            "DY_SIDECAR_USER_ID": f"{user_id}",
+            "DY_SIDECAR_PROJECT_ID": f"{project_id}",
+            "DY_SIDECAR_NODE_ID": f"{node_id}",
+        },
+    ), mock.patch.object(mounted_fs, "DY_VOLUMES", mock_dy_volumes):
+        print(os.environ)
+        yield
+
+
+# UTILS
+
+
+def _patch_settings_with_rabbit(app: FastAPI, rabbit_service: RabbitConfig) -> None:
+    app.state.settings.RABBIT_SETTINGS.__config__.allow_mutation = True
+    app.state.settings.RABBIT_SETTINGS.__config__.frozen = False
+
+    app.state.settings.RABBIT_SETTINGS.RABBIT_HOST = rabbit_service.host
+    app.state.settings.RABBIT_SETTINGS.RABBIT_PORT = rabbit_service.port
+    app.state.settings.RABBIT_SETTINGS.RABBIT_USER = rabbit_service.user
+    app.state.settings.RABBIT_SETTINGS.RABBIT_PASSWORD = rabbit_service.password
```

Comment:
is that really necessary? if you use either ```rabbit_service``` or monkeypatch?

Reply:
managed to fix this to avoid it